contracts/secret-storage/SecretManager.sol

```diff
@@ -633,19 +644,14 @@ contract SecretManager is
         uint256 _secretId,
         uint256 _lastAliveTimestamp
     ) internal returns (uint256 /* storageTimeDuration */) {
-        // uint256 secId = storeAckSecretIds[index];
         uint256 enclaveIndex = _getSelectedEnclaveIndex(_secretId, _enclaveAddress);
 
-        uint256 ackTimestamp;
-        if(_isReplacedStore(_secretId, enclaveIndex))
-            ackTimestamp = userStorage[_secretId].selectedEnclaves[enclaveIndex].replacedAckTimestamp;
-        else
-            ackTimestamp = userStorage[_secretId].ackTimestamp;
+        uint256 ackTimestamp = _getSecretStoreAckTimestamp(_secretId, enclaveIndex);
```

Comment:
ackTimestamp is also getting calculated twice, once here and once in the _updateUsdcDepositPostPayment, check it